06_puppy/pfannerj/syncstore.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+)
+
+// SyncStore is a sync.Map based in-memory implementation of PuppyStorer.
+type SyncStore struct {
+	sync.Map
```

Comment:
The point of using sync.Map is to protect your code from race conditions (make it "thread/goroutine safe".
You therefore need to also protect that no two goroutines can update your ID counter before using it.
You can use [Mutex](https://golang.org/pkg/sync/#Mutex) or [RWMutex](https://golang.org/pkg/sync/#RWMutex) for this (use lock at beginning of method and unlock at end where instructions need to be executed together).

Reply:
I was planning to make those changes as part of lab7... is it required for lab6 too?

Reply:
I checked the Lab7 readme... decided to leave this til Lab7... hope that's OK.

Reply:
I just noticed that somehow I have committed the old version (not updated with the fixes)... I will push again on Monday and then mark as 'ready for review'..